### ISSUE TITLE
[PROPOSAL]

### ISSUE THREAD

**Issue by lukepolo** [spatie/laravel-failed-job-monitor#3]
**In raising this issue, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide]().
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.
- [x] I have checked that the bug-fix I am reporting can be replicated.

---

### Description of the problem
Shouldn't we be able to to use laravel 5.1.* ? The 5.1 docs says the Queue::failing method exists. (which you are using) . Thoughts? 

### Steps to reproduce
composer update on Laravel 5.1

**Comment by freekmurze** [comment]
I believe the `Illuminate\Queue\Events\JobFailed`-class is new to Laravel 5.2.

**Comment by lukepolo** [comment]
ill double check that,ill post back in a few

**Comment by lukepolo** [comment]
Yes the event but in all reality its the same : 

5.1 failing function 
```
    /**
     * Register an event listener for the failed job event.
     *
     * @param  mixed  $callback
     * @return void
     */
    public function failing($callback)
    {
        $this->app['events']->listen('illuminate.queue.failed', $callback);
    }
```

5.2
```
/**
     * Register an event listener for the failed job event.
     *
     * @param  mixed  $callback
     * @return void
     */
    public function failing($callback)
    {
        $this->app['events']->listen(Events\JobFailed::class, $callback);
    }
```

**Comment by lukepolo** [comment]
blah it would be different sorry

**Comment by freekmurze** [comment]
Np, if you find a clean way to add Laravel 5.1 compatibility, I'll accept a PR.

**Comment by abbu1534** [comment]
I think Illuminate\Queue\Events\JobFailed class available in 5.2. Is there another way to use it in laravel 5.1

**Comment by TheShiftExchange** [comment]
@freekmurze - I had a similar solution already.

For 5.1 - you can do this in your AppServiceProviders.php file

`
public function boot()
	{
        Queue::failing(function ($connection, $job, $data) {
        	// send email here
        });
	}
`

So anytime a queue fails (for any reason) - it will do whatever you have set
